### PR TITLE
IMPRO-1606: Ensure cycle time updated in cycle blending

### PR DESCRIPTION
**Pull request by bayliffe**
In cycle blending the latest cube is currently used to set the cycle time (forecast reference time). If a file is missing for a particular validity time in the cycle in which the blending occurs, the cycle time for the output will be set as the most recent cycle for which data was available. This results in the forecast reference times for the blended output files from a given cycle potentially differing from one another.

There is an associated suite PR which adds the ```--cycletime``` argument for cycle blending.

We may wish to consider adding a CLI test for this behaviour, but I have not had time.

This work:
- Ensures that if a cycletime argument is provided, even if we are processing a single cube in cycle blending, the cycletime of the output is updated.
- I've added a unit test to cover this behaviour.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver/blending/calculate_weights_and_blend.py [hunk 1]
```diff
@@ -195,6 +196,7 @@ def process(self, cubelist, cycletime=None, model_id_attr=None,
             result = cube.copy()
             if attributes_dict is not None:
                 amend_attributes(result, attributes_dict)
+            result, = rebadge_forecasts_as_latest_cycle([result], cycletime)
```
Comment (gavinevans) [inline improver/blending/calculate_weights_and_blend.py]: Should the docstring above the `if` statement be updated, given that this is actually also catching the case where only one cycle has been provided for cycle blending, and the `forecast_reference_time` is being updated, in addition to updating the attributes?

Comment (bayliffe) [reply improver/blending/calculate_weights_and_blend.py]: Done.

**Comment by bayliffe** [comment]
@fionaRust as discussed, the multiple old cube case is covered by the suite change which now includes the cycletime argument. As such the blended output cycletime is updated in the WeightedBlendAcrossWholeDimension plugin. The change here covers the case of a single cube being passed in, which is returned before reaching this plugin.

File: improver_tests/acceptance/test_weighted_blending.py [hunk 2]
```diff
@@ -134,17 +134,36 @@ def test_percentile(tmp_path):
...
 
 
-@pytest.mark.slow
 def test_cycletime(tmp_path):
     """Test cycletime blending"""
     kgo_dir = acc.kgo_root() / "weighted_blending/cycletime"
     kgo_path = kgo_dir / "kgo.nc"
-    input_path = kgo_dir / "input.nc"
+    input_paths = sorted((kgo_dir.glob("input_temperature*.nc")))
+    print('inputs', input_paths)
```
Comment (fionaRust) [inline improver_tests/acceptance/test_weighted_blending.py]: Need to remove this print statement. Also I think you need to exclude one of the input cubes or change the cycletime as one of the input cubes has the same cycletime as the cycletime you pass in.

File: improver_tests/acceptance/test_weighted_blending.py [hunk 3]
```diff
@@ -134,17 +134,54 @@ def test_percentile(tmp_path):
...
+    returned cube is user specified."""
+    kgo_dir = acc.kgo_root() / "weighted_blending/cycletime"
+    kgo_path = kgo_dir / "kgo_specified_frt.nc"
+    input_paths = sorted((kgo_dir.glob("input_temperature*.nc")))
+    input_paths.pop(-1)
+    output_path = tmp_path / "output.nc"
+    args = ["--coordinate", "forecast_reference_time",
+            "--y0val", "1.0",
+            "--ynval", "4.0",
+            "--cycletime", "20200218T0500Z",
```
Comment (fionaRust) [inline improver_tests/acceptance/test_weighted_blending.py]: Did you mean this to be T0500Z or T0600Z?

Comment (bayliffe) [reply improver_tests/acceptance/test_weighted_blending.py]: Not sure how I managed that. 0600Z is what I was after. Altered and KGO updated.

**Pull request merged**